tests/behaviour/util.rs

```diff
@@ -19,8 +19,158 @@
+    answer: &ConceptMap,
+) -> bool {
+    stream::iter(answer_identifiers.keys())
+        .all(|key| async {
+            answer.map.contains_key(key.clone())
+                && match_answer_concept(context, answer_identifiers.get(key).unwrap(), answer.get(key).unwrap()).await
+        })
+        .await
+}
+
+pub async fn match_answer_concept(context: &Context, answer_identifier: &String, answer: &Concept) -> bool {
+    let identifiers: Vec<&str> = answer_identifier.splitn(2, ":").collect();
+    match identifiers[0] {
+        "key" => key_values_equal(context, identifiers[1], answer).await,
+        "label" => labels_equal(identifiers[1], answer),
+        "value" => values_equal(identifiers[1], answer),
+        _ => unreachable!(),
+    }
+}
+
+async fn key_values_equal(context: &Context, expected_label_and_value: &str, answer: &Concept) -> bool {
+    let identifiers: Vec<&str> = expected_label_and_value.splitn(2, ":").collect();
+    assert_eq!(identifiers.len(), 2, "Unexpected table cell format: {expected_label_and_value}.");
+
+    let res = match answer {
+        Concept::Entity(entity) => {
+            async { entity.get_has(context.transaction(), vec![], vec![Annotation::Key]) }
+                .and_then(|stream| async { stream.try_collect::<Vec<_>>().await })
+                .await
+        }
+        Concept::Attribute(attr) => {
+            async { attr.get_has(context.transaction(), vec![], vec![Annotation::Key]) }
+                .and_then(|stream| async { stream.try_collect::<Vec<_>>().await })
+                .await
+        }
+        Concept::Relation(rel) => {
+            async { rel.get_has(context.transaction(), vec![], vec![Annotation::Key]) }
+                .and_then(|stream| async { stream.try_collect::<Vec<_>>().await })
+                .await
+        }
+        _ => unreachable!("Unexpected Concept type: {answer:?}"),
+    };
+    match res {
+        Ok(_) => {
+            let equals = res
+                .unwrap()
+                .into_iter()
+                .filter(|attr| attr.type_.label == identifiers[0])
+                .collect::<Vec<_>>()
+                .first()
+                .and_then(|attr| Some(value_equals_str(&attr.value, identifiers[1])));
+            match equals {
+                Some(val) => val,
+                None => false,
+            }
+        }
+        Err(_) => false,
+    }
+}
+
+fn labels_equal(expected_label: &str, answer: &Concept) -> bool {
+    answer.type_label_cloned() == expected_label
+}
+
+fn values_equal(expected_label_and_value: &str, answer: &Concept) -> bool {
+    let identifiers: Vec<&str> = expected_label_and_value.splitn(2, ":").collect();
+    assert_eq!(identifiers.len(), 2, "Unexpected table cell format: {expected_label_and_value}.");
+    let Concept::Attribute(Attribute { value, .. }) = answer else { unreachable!() };
+    value_equals_str(value, identifiers[1])
+}
+
+fn value_equals_str(value: &Value, expected: &str) -> bool {
```

Comment:
heads up: these will have to be updated once we introduce `value` and `attr` separately from the rebase off `master`!